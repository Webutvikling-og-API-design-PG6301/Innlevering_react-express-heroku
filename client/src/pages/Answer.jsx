import React from "react";
import { Route, Routes } from "react-router";
import { Link } from "react-router-dom";
const Answer = ({ isRightAnswer, isAnsweredQuestion }) => {
  return (
    <div>
      <Routes>
        <Route path={"correct"} element={<h1>Correct!</h1>} />
        <Route path={"wrong"} element={<h1>Wrong!</h1>} />
      </Routes>

      <h2>
        Your score is {isRightAnswer} / {isAnsweredQuestion}
      </h2>
      <Link to="/">
        <button>Return to home</button>
      </Link>
      <Link to="/question">
        <button>New quiz</button>
      </Link>
    </div>
  );
};

export default Answer;
