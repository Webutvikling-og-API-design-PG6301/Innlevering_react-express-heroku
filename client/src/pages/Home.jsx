import React from "react";
import { Link } from "react-router-dom";
const Home = () => {
  return (
    <div>
      <h1>Home page</h1>

      <Link to="/question">
        <button>New quiz</button>
      </Link>
      <Link to="/answer/*">
        <button>Show anwers</button>
      </Link>
    </div>
  );
};

export default Home;
