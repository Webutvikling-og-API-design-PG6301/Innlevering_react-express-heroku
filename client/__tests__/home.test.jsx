import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router-dom";
import Answer from "../src/pages/Answer";
import Home from "../src/pages/Home";

describe("Quiz application", () => {
  it("should show home component", () => {
    const element = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>,
      element
    );
    expect(element.innerHTML).toMatchSnapshot();
  });

  it("should show answers", () => {
    const element = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <Answer isRightAnswer={3} isAnsweredQuestion={5} />
      </MemoryRouter>,
      element
    );
    expect(element.innerHTML).toMatchSnapshot();
  });
});
