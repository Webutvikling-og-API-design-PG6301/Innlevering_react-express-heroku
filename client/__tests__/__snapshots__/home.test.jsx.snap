// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quiz application should show answers 1`] = `"<div><h2>Your score is 3 / 5</h2><a href=\\"/\\"><button>Return to home</button></a><a href=\\"/question\\"><button>New quiz</button></a></div>"`;

exports[`Quiz application should show home component 1`] = `"<div><h1>Home page</h1><a href=\\"/question\\"><button>New quiz</button></a><a href=\\"/answer/*\\"><button>Show anwers</button></a></div>"`;
